_content/tour/rus/interfaces.article

```diff
@@ -34,177 +34,179 @@ maintainability.
-also help my code decouple itself from change by asking for concrete data based
-on what it can do. It’s not limited to one type of data.
+Интерфейсы позволяют группировать конкретные данные по тому, что они могут делать. Здесь мы
+фокусируемся на том, что данные могут делать, а не о том, что это эти данные из себя
+представляют. Интерфейсы также помогают минимизировать изменения кода, работая с данными
+на основе того что они могут выполнять. Это не ограничивается одним единственным типом данных.
 
-You must do my best to understand what data changes are coming and use interfaces
-to decouple my program from that change. Interfaces should describe behavior and
-not state. They should be verbs and not nouns.
+Вам нужно постараться понять, какие ожидаются изменения данных, и с помощью интерфейсов
+защитить программный код от этих изменений. Интерфейсы должны описывать поведение,
+а не состояние. Они должны быть глаголами, а не существительными.
 
-Generalized interfaces that focus on behavior are best. Interfaces with more than
-one method have more than one reason to change. Interfaces that are based on nouns,
-tend to be less reusable, are more susceptible to change, and defeat the purpose
-of the interface. Uncertainty about change is not a license to guess but a directive
-to STOP and learn more. You must distinguish between code that defends against
-fraud vs protects against accidents.
+Наилучший выбор - это обобщённые интерфейсы, которые фокусируются на поведении. Интерфейсы с
+более чем одним методом имеют больше причин для изменения. Интерфейсы, основанные на существительных,
+обычно сложнее использовать повторно, они больше подвержены изменениям и противоречат основному
+назначению интерфейсов. Неопределенность в части ожидаемых изменений - это не повод строить догадки,
+а указатель на необходимость остановиться и узнать больше. Вам следует различать код защищающий
+от мошенничества, и код, защищающий от несчастных случаев.
 
-Use an interface when:
+Используйте интерфейс, когда:
 
-- Users of the API need to provide an implementation detail.
-- API’s have multiple implementations they need to maintain internally.
-- Parts of the API that can change have been identified and require decoupling.
+- Пользователям API нужно предоставить детали реализации.
+- API необходимо поддерживать несколько внутренних реализаций.
+- Вы обнаружили части API, которые могут измениться и хотите их разделить.
 
-Don't use an interface:
-    
-- For the sake of using an interface.
-- To generalize an algorithm.
-- When users can declare their own interfaces.
-- If it's not clear how the interface makes the code better.
+Не используйте интерфейс:
+
+- Просто ради использования интерфейса.
+- Для обобщения алгоритма.
+- Когда пользователи могут объявить свои собственные интерфейсы.
+- Если не ясно, как интерфейс улучшает код.
 
-** Interfaces Are Valueless
+** Интерфейсы не содержат значения
```

Comment:
```suggestion
** Интерфейсы не работают со значениям
```
Here I struggle to find a translation, maybe you have a better idea.
I think "не содержат" is a bit inaccurate because if we talk about an internal representation of interface, we also talk about values (like in https://research.swtch.com/interfaces).